server/internal/server/server.go

```diff
@@ -0,0 +1,155 @@
+package server
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"time"
+
+	"github.com/go-logr/logr"
+	v1 "github.com/llmariner/api-usage/api/v1"
+	"github.com/llmariner/api-usage/server/internal/config"
+	"github.com/llmariner/api-usage/server/internal/store"
+	"github.com/llmariner/rbac-manager/pkg/auth"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/health"
+	"google.golang.org/grpc/health/grpc_health_v1"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/reflection"
+	"google.golang.org/grpc/status"
+)
+
+const (
+	defaultOrgnizationID = "default"
+	defaultProjectID     = "default"
+	defaultClusterID     = "default"
+	defaultTenantID      = "default-tenant-id"
+)
+
+// New creates a new server.
+func New(store *store.Store, logger logr.Logger) *Server {
+	return &Server{
+		store:  store,
+		logger: logger.WithName("server"),
+	}
+}
+
+// Server is the server for the api-usage service.
+type Server struct {
+	v1.UnimplementedAPIUsageServiceServer
+
+	srv    *grpc.Server
+	store  *store.Store
+	logger logr.Logger
+}
+
+// Run starts the server.
+func (s *Server) Run(ctx context.Context, port int, authConfig config.AuthConfig) error {
+	s.logger.Info("Starting gRPC server...", "port", port)
+
+	var opt grpc.ServerOption
+	if authConfig.Enable {
+		ai, err := auth.NewInterceptor(ctx, auth.Config{
+			RBACServerAddr: authConfig.RBACInternalServerAddr,
+			AccessResource: "api.api_usages",
```

Comment:
You might already know, but https://github.com/llmariner/rbac-manager/blob/main/deployments/rbac-server/values.yaml#L33 needs to be updated.